doc/release/upcoming_changes/25434.change.rst

```diff
@@ -0,0 +1,15 @@
+Cleanup of initialization of ``numpy.dtype`` with strings with commas
```

Comment:
I guess the `(2,)` deprecation may be a bit because we only trigger this parsing when there is a comma somewhere?
Anyway, I am a bit ambivalent to it, but also don't worry about it (someone can still be annoyed and undo it anyway).

Reply:
Yes, the path only gets hit if there is a comma in the string. If "(2)i" was normally acceptable, I would definitely have let it be, but since it is not, I think we might as well go for slightly more logical. And I agree we can undo it if it causes too many problems.